server/chess.py

```diff
@@ -100,16 +100,17 @@ def hypothetical_check(
                 typing.Tuple[models.Piece, int, int], ...
             ]) -> bool:
         """Check if a series of moves would put a side in check."""
-        if self.hypothetical_moves is None:
+        if moves is None:
```

Comment:
```suggestion
        if self.hypothetical_moves is not None:
```
I forgot a `not`. The purpose of this check is to ensure that `hypothetical_check` does not call itself.